### DISCUSSION TITLE
file_picker_in_current_buffer_directory not working

### DISCUSSION THREAD

**Discussion by BobbyByrne** [helix-editor/helix#8816] [Troubleshooting]
### Summary

This command doesn't do anything. It simply closes the command panel.

The PR for this feature has another user experiencing the same issue back in April. 
https://github.com/helix-editor/helix/pull/4666#issuecomment-1509037354

I'm also experiencing the same issue with `<space>f` and `<space>F` doing the same thing. 

### Reproduction Steps

 

I tried this:

1. Opened helix in a project root dir. example `/projects/foo`
2. Opened a file buffer in a nested dir `/projects/foo/bar/baz.rs`
3. Attempted all three commands starting with `file_picker`

I expected this to happen:

- `file_picker`
  - ?
- `file_picker_in_current_directory`
  - I would expect the file picker to open in `/projects/foo`
- `file_picker_in_current_buffer_directory`
  - I would expect the file picker to open in `/projects/foo/bar`
  - I think I would expect this to be the same behavior as `file_picker`, but I'm not sure what the actual default behavior is.

Instead, this happened:

- `file_picker` using space f
  - opens file picker in `/projects/foo`
- `file_picker_in_current_directory` using space F
  - opens file picker in `/projects/foo`
- `file_picker` using command window `<space>?`
  - closes command window and does nothing...
- `file_picker_in_current_directory` using command window `<space>?`
  - closes command window and does nothing...
- `file_picker_in_current_buffer_directory` using command window `<space>?`
  - closes command window and does nothing...

### Platform

Windows (WSL)

### Terminal Emulator

Windows Terminal Version: 1.18.2822.0

### Installation Method

releases page

### Helix Version

helix 23.10 (f6021dd0)

**Comment by gabydd** [comment]
The pickers not working from the command palette is: https://github.com/helix-editor/helix/issues/4508 if you want the picker in current buffer directory then at the moment you need to bind it to key

**Reply by jerabaul29** [reply]
Could you provide a snippet config example of how to do this, in particular, how to / is it possible to bind it to a "space mode shortcut", and make it visible in the "Space" documentation popup? I would like to bind it to "space" (at which point the space popup appears, and it would be great if the new o option get listed) + "o".

**Reply by jerabaul29** [reply]
Ok, found it here: https://github.com/helix-editor/helix/issues/3052 . The config entry to add is:

```
[keys.normal."space"]
o = "file_picker_in_current_buffer_directory"
```

(credits @tennix :) ).

This works perfectly fine, and the 'o' option gets displayed at the bottom of the "space" self documentation popup :) .

**Comment by tennox** [comment]
To clarify the difference (as far as I understand):
- **`file_picker_in_current_directory`** - will always be the directory where helix was opened from
- **`file_picker`** - is based on the root of the project (e.g. in a monorepo, if you opened hx in a subdir)... but I'm not sure what this is detected by (git root?)
